### PR TITLE
engine: add some comments

### PR DESCRIPTION
**Pull request by SparkYuan**
1. re-organization engine pkgs
2. add some comments in the pkg engine to make codes easier to read

### CODE REVIEW

**Comment by SparkYuan** [comment]
I have read the CLA Document and I hereby sign the CLA

File: pkg/engine/operation/apply.go [hunk 1]
```diff
@@ -44,6 +44,11 @@ func NewApplyGraph(m *models.Spec, priorState *states.State) (*dag.AcyclicGraph,
 	return graph, s
 }
 
+// Apply means turn all actual infra resources into the desired state described in the request by invoking a specified Runtime.
+// Like other operations, Apply has 3 main steps during the whole process.
+// 1. parse resources and their relationship to build a DAG and should take care of those resources that will be deleted
+// 2. walk this DAG and execute all graph nodes concurrently, besides the entire process should follow dependencies in this DAG
+// 3. during the execution of each node, it will invoke different runtime according to the resource type
```
Comment (chai2010) [inline pkg/engine/operation/apply.go:51]: the go doc do not support markdown format, the list item should split by blank line.
see https://pkg.go.dev/fmt

Comment (SparkYuan) [reply pkg/engine/operation/apply.go:51]: these lines are just nomarl comments for the codes below....

File: pkg/engine/doc.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+//
+// Kusion Engine is a middle layer between Spec and the actual infrastructure.
+// The major function of this engine is to parse the KCL compiled result and to turn all actual infra resources into the desired state
+// described in KCL. It consists of 3 parts:
+// 1. DAG engine: parse resources in the Spec and figure out which resource should be created,
+// updated or deleted. The analysis result will be converted into a DAG.
+// 2. Runtime: an interface that represents an actual infrastructure.
+// All operations that trying to manipulate a resource should delegate to one Runtime
+// 3. State: state is a record of an operation's result.
+// It is often used as a datasource for 3-way merge/diff in operations like Apply or Preview
```
Comment (ldxdl) [inline pkg/engine/doc.go:11]: + Kusion Engine is a middle layer between API spec and the actual infrastructure. And Who will use the Engine?
+ Remove 'KCL' in the doc.  API spec does not need to perceive the upper layer input.
+ API Spec should be versioned, and should be backwards compatible?
+ The conversion mechanism, codec, convention.. from api spec to internal model?
+ I prefer to split 'DAG engine' to a diff engine, a DAG builder and an execution engine?
+ Runtime: runtime is usually not an interface 
+ Important interface, structs and usage examples?

Comment (ldxdl) [reply pkg/engine/doc.go:11]: Judging from the code I have learned so far and his positioning, maybe this code structure would be more appropriate:

/types -- engine‘s internal types
/models -- engine's internal model
/exec -- nodes exec flow
/dag -- dag builder, walker, etc
/diff -- diff tools
/runtime -- runtime apis and impls
/state -- state apis and impls
api.go -- engine's core public interfaces and functionalities.
types.go -- entine's public types.

Through such code organization, the core functions of the engine can be better showed, and the codes of different functional modules can also be reasonably organized.

You can modify it as you like. Hope the advice helps.

File: pkg/engine/operation/apply.go (hunk 1 above)
Comment (elliotxx) [reply pkg/engine/operation/apply.go:51]: Comment need to follow official specifications:
https://go.dev/doc/comment

File: pkg/kusionctl/cmd/apply/options.go [hunk 3]
```diff
@@ -8,12 +8,16 @@ import (
 	"strings"
 	"sync"
 
+	"kusionstack.io/kusion/pkg/engine/operation"
+	opsmodels "kusionstack.io/kusion/pkg/engine/operation/models"
```
[the remark this answers is not in the document]
Comment (SparkYuan) [reply pkg/kusionctl/cmd/apply/options.go:12]: ok, I have added a doc file in pkg/engine/operation/models and I will make this pkg internal later

**Review by elliotxx — APPROVED** [review]
lgtm

**Pull request merged**